### ISSUE TITLE
Photo view: significant faster than on android

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/ios#583]
Your photo view is significant faster than on android.
On android we are using webdav search, which then gets back all images on server (on big instances, this even might take ~15s for 10k images), after that we fetch the metadata info for this files, and after that we have to list them (which is the fastest part).
So all in all this takes 15-20s for 5k images.

What and how are you doing that it is this amazing fast?

**Comment by marinofaggiana** [comment]
oh! Nothing magic, I use webdav search .. maybe the difference is after the download (XML parser ?)
bui I'm not happy with iOS performance :-))) , now I have detect a new XML parser write in C for speed up (I have to find the time to get it) .....  

Note : I use multi-tread for download thumbnails, parse the XML, setting the metadata into DB and for view the images ... 

@tobiasKaminsky  I do not know if it helps you

**Comment by tobiasKaminsky** [comment]
Thank you for your reply @marinofaggiana.
Are you refreshing every time you enter the photo view?

Can you give me some detail:
- how long does the search query take for 5000k images?
- how long does the parsing take?
- how long takes the metadata step?
- how long for viewing/downloading the images?

Thank you :-) :+1:

**Comment by marinofaggiana** [comment]
No, I refresh (webdav search) only when the app go back in active mode and the etag of root folder (/ or the root folder selected) is changed but i refresh the DB when enter in photo view (this is important for the file uploaded via App).

Before the detail time : 

1 - when the App is activated I start the query
2 - I pass the query at XML engine
3 - I insert all in local DB

--- end

When I enter in photo view I download the thumbnails

**Comment by marinofaggiana** [comment]
@tobiasKaminsky sorry for the delay, tomorrow i make a test.

**Comment by marinofaggiana** [comment]
@tobiasKaminsky  test for 20.000 item (Images/videos)

Download Search webDAV = 01.15.00 (not important, ADSL, Server speed etc.)

XML engine = 00.04.00
Creation internal metadata = 00.15.00

Save on internal DB = 01.30.00  --> 45 sec. for loop archive for remove item no longer present  (mmm no good :-)) )  and 45 sec. for insert/update the 20.000 item

Prepare for view and view = 00.00.01

Cheers
m.

**Comment by marinofaggiana** [comment]
@tobiasKaminsky close this ?
